lib/functions/main/config-prepare.sh

```diff
@@ -220,6 +220,12 @@ function config_post_main() {
 	# So we gotta explictly know the major.minor to be able to do that scheme.
 	# If we don't know, we could use BRANCH as reference, but that changes over time, and leads to wastage.
 	if [[ "${skip_kernel:-"no"}" != "yes" ]]; then
+
+		# call hooks to do late validation/mutation of sources, branches, patch dirs, etc.
+		call_extension_method "late_family_config" <<- 'LATE_FAMILY_CONFIG'
+			*late defaults/overrides, main hook point for KERNELSOURCE/BRANCH and BOOTSOURCE/BRANCH etc*
```

Comment:
Do we really need another hook function? Can't we do the same thing with post_family_confg function. I get the validation part. But why mutate things now?

Reply:
A separate hook is required, otherwise ordering the hooks would be a huge mess.
It also helps to understand that this logic is _shared_ and _global_ across all families, and not part of a single family's handling (which is what post_family_config is for).

Reply:
Sorry I failed to see why its required. As I understand, we load the family source file first, then common and then arch specific one. So we do already have common.conf and <arch>.conf files to set defaults. Why do we need another hook?

Could you please share the use case so that I can try to see what exactly you are trying to solve by adding another hook function?

Reply:
The case: centralized, non-family specific, management of mainline kernel versions.
The `late_family_config` hook was introduced to keep `post_family_config` logic and ordering unaltered but do global changes to all families / kernels / even vendor ones.
The `mainline_kernel_decide_version` is _specific_ to mainline kernels.